src/cli/commands/pin/ls.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+module.exports = {
+  command: 'ls',
+
+  describe: 'List objects pinned to local storage.',
+
+  builder: {
+    path: {
+      type: 'string',
+      describe: 'List pinned state of specific <ipfs-path>.'
+    },
+    type: {
+      type: 'string',
+      alias: 't',
+      default: 'all',
+      describe: ('The type of pinned keys to list. ' +
+                 'Can be "direct", "indirect", "recursive", or "all".')
+    },
+    quiet: {
+      type: 'boolean',
+      alias: 'q',
+      default: false,
+      describe: 'Write just hashes of objects.'
+    }
+  },
+
+  handler: (argv) => {
+    const paths = argv.path && argv.path.split(' ')
+    const type = argv.type
+    const quiet = argv.quiet
+    argv.ipfs.pin.ls(paths, { type, quiet })
+      .then((pinset) => {
+        pinset.forEach((item) => {
+          let line = item.hash
+          if (!quiet) {
+            line += ` ${item.type}`
+            if (item.through) {
+              line += ` through ${item.through}`
+            }
+          }
+          console.log(line)
+        })
+      })
```

Comment:
Either `.catch` a possible error or use the callback API so that you don't forget :)